dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheClientGauges.java

```diff
@@ -35,28 +43,60 @@
 @SuppressWarnings("UnnecessaryLambda")
 final class ApacheClientGauges {
 
-    private static final List<WeakReference<PoolingHttpClientConnectionManager>> pools = new CopyOnWriteArrayList<>();
```

Comment:
maybe I don't fully understand the context about why we had a static field here in the first place, but could we do something like the `weakGauge` thing we just added in ConcurrencyLimitedChannel? https://github.com/palantir/dialogue/blob/7dc6e7cfbc508231220ab3b2f108445ac690c24b/dialogue-core/src/main/java/com/palantir/dialogue/core/ConcurrencyLimitedChannel.java#L153-L174

Reply:
This utility is built to capture the sum of values when we roll over. I think those statistics are important for tracking connection pools, where we don't want to replace the existing value outright.

Reply:
got it makes sense - I think I hadn't fully understood the purpose of this class initially, but it makes sense now.